cli/commands.go

```diff
@@ -2,44 +2,89 @@ package cli
 
 import (
 	"fmt"
-	"os"
+	"regexp"
 	"strings"
 
 	"github.com/rokka-io/rokka-go/rokka"
 )
 
+type UnknownCommandError string
+
+func (e UnknownCommandError) Error() string {
+	return string(e)
+}
+
 type Command struct {
-	args        []string
-	description string
-	fn          func(*rokka.Client)
+	Args        []string
+	Description string
+	fn          func(*rokka.Client, []string, map[string]string) error
```

Comment:
how does a CLI user know which options are possible? What if a user specifies options which are not available for the chosen command?

Reply:
Addressed in original PR message.